src/infer_request.cc

```diff
@@ -1134,6 +1135,24 @@ InferenceRequest::Normalize()
                 implicit_batch_note);
       }
     }
+    // Matching incoming request's shape and byte size to make sure the
+    // payload contains correct number of elements
+    {
+      const size_t& byte_size = input.Data()->TotalByteSize();
+      const auto& data_type = input.DType();
+      const auto& input_dims = *shape;
+      const int64_t expected_byte_size =
+          triton::common::GetByteSize(data_type, input_dims);
```

Comment:
Note that this API for `TYPE_STRING` returns 0 and we might some custom handling for that.

Reply:
Fixed